tests/testsuite/registry.rs

```diff
@@ -520,10 +520,11 @@ fn relying_on_a_yank_is_bad() {
         .with_status(101)
         .with_stderr_contains(
             "\
-error: no matching version `= 0.0.2` found for package `baz`
-location searched: registry `[..]`
-versions found: 0.0.1
+error: failed to select a version for `baz` satisfying the requirement `= 0.0.2`
+  candidate versions found which didn't match: 0.0.1
+  location searched: `[..]` index (which is replacing registry `[..]`)
 required by package `bar v0.0.1`
+    ... which is depended on by `foo [..]`
```

Comment:
Why did this change? I am not objecting, just mechanically, what changed making us print `which is depended on by`?

Reply:
Oh I just expanded the error message a bit here, only a subset of the output was previously tested with `with_stderr_contains` (vs `with_stderr`)

Reply:
Ah. 👍